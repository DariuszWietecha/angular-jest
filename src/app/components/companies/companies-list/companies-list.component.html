<div class="text-right">
	<a [routerLink]="['/companies/create/']" class="btn btn-primary">Create</a>
</div>

<h4>Companies</h4>
<div class="d-flex flex-wrap">
	<app-company *ngFor="let company of companies" [company]="company"></app-company>
</div>