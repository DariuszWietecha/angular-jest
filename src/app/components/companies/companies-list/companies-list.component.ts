import { Component, OnInit} from '@angular/core';
import { CompaniesService, ICompany } from '../../../services/companies.service';
import { Input } from '@angular/core';

@Component({
  selector: 'app-companies-list',
  templateUrl: './companies-list.component.html',
  styleUrls: ['./companies-list.component.scss']
})
export class CompaniesListComponent implements OnInit  {
  @Input() companies: ICompany[];

  constructor(private companiesService: CompaniesService) { }

  ngOnInit() {
    this.companiesService.list()
      .subscribe((data) => {
        this.companies = data;
      });
}

}
