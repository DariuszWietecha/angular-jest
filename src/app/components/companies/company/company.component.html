<div class="card-body">
  <h4 class="card-title">{{ company.name }}</h4>

  <dt>Name:</dt>
  <dd>{{ company.name }}</dd>

  <dt>Logo:</dt>
  <dd>{{ company.logoUrl }}</dd>

  <dt>E-mail:</dt>
  <dd>{{ company.email }}</dd>

  <dt>Categories:</dt>
  <dd>
    <ul>
      <li *ngFor="let category of company.categories">{{category.name}}</li>
    </ul>
  </dd>

  <a [routerLink]="['/companies/edit', company.id]" class="btn btn-primary"  *ngIf="showEditButton === true">Edit</a>
</div>