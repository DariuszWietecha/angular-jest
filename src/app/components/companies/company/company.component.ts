import { Component, Input, HostBinding } from '@angular/core';
import { ICompany } from '../../../services/companies.service';

@Component({
  selector: 'app-company',
  templateUrl: './company.component.html',
  styleUrls: ['./company.component.scss']
})

export class CompanyComponent {
  @HostBinding('class') className = 'card col-4';
  @Input() company: ICompany;
  @Input() showEditButton = true;
  constructor() { }
}
