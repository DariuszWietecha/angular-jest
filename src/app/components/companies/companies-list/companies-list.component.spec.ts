import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { CompaniesService } from '../../../services/companies.service';
import { CompaniesListComponent } from './companies-list.component';
import { CompanyComponent } from '../company/company.component';
import { HttpClientModule } from '@angular/common/http';

describe('CompaniesListComponent', () => {
  let component: CompaniesListComponent;
  let fixture: ComponentFixture<CompaniesListComponent>;
  const comapaniesListMock = require('../../../../../test/data/companies/companies-get.json');

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientModule, RouterTestingModule],
      declarations: [CompanyComponent, CompaniesListComponent],
      providers: [CompaniesService],
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CompaniesListComponent);
    component = fixture.componentInstance;
    component.companies = comapaniesListMock;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapShot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
