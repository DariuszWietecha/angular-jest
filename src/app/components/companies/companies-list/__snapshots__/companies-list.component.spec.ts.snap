// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompaniesListComponent snapShot 1`] = `
<app-companies-list
  companies={[Function Array]}
  companiesService={[Function CompaniesService]}
>
  <div
    class="text-right"
  >
    <a
      class="btn btn-primary"
      href="/companies/create"
      ng-reflect-router-link="/companies/create/"
    >
      Create
    </a>
  </div><h4>
    Companies
  </h4><div
    class="d-flex flex-wrap"
  >
    
    <app-company
      class="card col-4"
      ng-reflect-company="[object Object]"
    >
      <div
        class="card-body"
      >
        <h4
          class="card-title"
        >
          Great Company
        </h4>
        <dt>
          Name:
        </dt>
        <dd>
          Great Company
        </dd>
        <dt>
          Logo:
        </dt>
        <dd>
          Great cube
        </dd>
        <dt>
          E-mail:
        </dt>
        <dd>
          mail@greatcompany.com
        </dd>
        <dt>
          Categories:
        </dt>
        <dd>
          <ul>
            
            <li>
              Big Category
            </li>
          </ul>
        </dd>
        
        <a
          class="btn btn-primary"
          href="/companies/edit/f0a6d4b0-6591-11ea-a253-fd7829c2ab13"
          ng-reflect-router-link="/companies/edit,f0a6d4b0-6591-"
        >
          Edit
        </a>
      </div>
    </app-company>
    <app-company
      class="card col-4"
      ng-reflect-company="[object Object]"
    >
      <div
        class="card-body"
      >
        <h4
          class="card-title"
        >
          Small Company
        </h4>
        <dt>
          Name:
        </dt>
        <dd>
          Small Company
        </dd>
        <dt>
          Logo:
        </dt>
        <dd>
          Small ball
        </dd>
        <dt>
          E-mail:
        </dt>
        <dd>
          mail@smallcompany.com
        </dd>
        <dt>
          Categories:
        </dt>
        <dd>
          <ul>
            
            <li>
              Quite big company
            </li>
          </ul>
        </dd>
        
        <a
          class="btn btn-primary"
          href="/companies/edit/f0a6d4b0-6591-11ea-a253-fd7829c2aaaa"
          ng-reflect-router-link="/companies/edit,f0a6d4b0-6591-"
        >
          Edit
        </a>
      </div>
    </app-company>
  </div>
</app-companies-list>
`;
